src/redux/selectors/proxies.js

```diff
@@ -2,6 +2,31 @@ import { createSelector } from 'reselect';
 import access from './stateAccessor';
 
 // ----------------------------------------------------------------------------
+// Convenience methods to traverse through a proxies property and ui arrays
+// to update the collapsed state of the various collapsible groups that may be
+// present.
+// ----------------------------------------------------------------------------
+
+function checkPropertyList(propList, uiList, groupCollapseState) {
+  propList.forEach((prop, idx) => {
+    const ui = uiList[idx];
+    const id = [prop.id, prop.name].join(':');
+    if (id in groupCollapseState) {
+      prop.value = !groupCollapseState[id];
+    }
+
+    if (prop.children) {
+      checkPropertyList(prop.children, ui.children, groupCollapseState);
+    }
+  });
+}
+
+function decorateCollapsibleGroups(proxy, groupCollapseState) {
+  checkPropertyList(proxy.properties, proxy.ui, groupCollapseState);
+  return proxy;
+}
+
+// ----------------------------------------------------------------------------
```

Comment:
As you can see processing the data that way was not too bad instead of doing in the widget...